controller/src/main/java/io/pravega/controller/store/stream/tables/TableHelper.java

```diff
@@ -427,7 +432,13 @@ public static int getSegmentCount(final byte[] segmentTable) {
     public static boolean isScaleOngoing(final byte[] historyTable, final byte[] segmentTable) {
 
         HistoryRecord latestHistoryRecord = HistoryRecord.readLatestRecord(historyTable, false).get();
-        return latestHistoryRecord.isPartial() || !latestHistoryRecord.getSegments().contains(getLastSegmentNumber(segmentTable));
+        // Either last row of HistoryTable is partial,
+        // or negation of the following condition.
+        // Non-empty set of segments in last row of HistoryTable implies that
+        // those segments contain highest numbered segment from SegmentTable.
+        return latestHistoryRecord.isPartial() ||
```

Comment:
This check is not redundant. Consider the state of SegmentTable and HistoryTable after seal completion. SegmentTable has an additional SealSentinal, but the last record of HistoryTable has empty segment list. In this state, the highest numbered segment in SegmentTable is not present in the last record of HistoryTable, although no scaling operation in ongoing.